plugins/node/opentelemetry-koa-instrumentation/src/utils.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AttributeNames,
+  KoaContext,
+  KoaComponentName,
+  KoaMiddleware,
+  KoaLayerType,
+} from './types';
+import { Attributes } from '@opentelemetry/api';
+
+export const getMiddlewareMetadata = (
+  context: KoaContext,
+  layer: KoaMiddleware,
+  isRouter: boolean,
+  layerPath?: string
+): {
+  attributes: Attributes;
+  name: string;
+} => {
+  if (isRouter) {
```

Comment:
Would rather just have one function for router and another function for whatever not router is instead of this boolean switching.

Reply:
This would require boolean switching on `isRouter` in the `_patchLayer` function of koa.ts - right now, the `_patchLayer` function is designed to instrument both types of middleware smoothly. I can definitely change to checking `isRouter` there, but I think it might be a bit less clean than deferring it to here in utils